### PR TITLE
Fixed cosmos instrumentation code break if invalid credentials are given

### PR DESCRIPTION
**Pull request by Angith**

### CODE REVIEW

File: example/cosmos/go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.0
 	github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos v0.3.6
 	github.com/google/uuid v1.4.0
-	github.com/instana/go-sensor v1.61.0
+	github.com/instana/go-sensor v1.62.0
```
Comment (nithinputhenveettil) [inline example/cosmos/go.mod:9]: Please rebase with the latest `main`.

Comment (Angith) [reply example/cosmos/go.mod:9]: Done 👍

**Review by nithinputhenveettil — APPROVED** [review]
Good work, man! :shipit:

File: instrumentation/instacosmos/container_client.go [hunk 2]
```diff
@@ -534,9 +534,11 @@ func (icc *instaContainerClient) setError(s tracing.Span, err error) {
 	})
 
 	// setting status code of error response
-	var responseErr *azcore.ResponseError
-	errors.As(err, &responseErr)
-	icc.setStatus(s, responseErr.StatusCode)
+	var responseErr = new(azcore.ResponseError)
+	ok := errors.As(err, &responseErr)
+	if ok {
```
Comment (sanojsubran) [inline instrumentation/instacosmos/container_client.go:539]:
```suggestion
	if ok := errors.As(err, &responseErr); ok { 
```

Suggestion/Nitpicking: You can combine the `if` blocks for simple cases if you wish to

**Review by sanojsubran — APPROVED** [review]
Looks good!

**Pull request merged**